Resources/public/js/views/services/ez-roleserversideviewservice.js

```diff
@@ -0,0 +1,281 @@
+                service = this;
+
+            this._assignRoleNotificationStarted(
+                data.roleId, data.roleName, e.selection
+            );
+
+            userService.loadRole(data.roleId, function (error, response) {
+                var tasks = new Y.Parallel(),
+                    end = tasks.add(function (err, response) {
+                        if (err) {
+                            service._error('tasks.add() failed: ' + err.message);
+                            return;
+                        }
+                    });
+
+                if (error) {
+                    service._error('loadRole failed: ' + error.message);
+                    return;
+                }
+
+                service._assignRoleToEachAssignee(e, response.document.Role, end);
+
+                tasks.done(function () {
+                    service._assignRoleCallback(data.roleId, data.roleName, e.selection,
+                        Y.bind(data.afterUpdateCallback, service));
+                });
+            });
+        },
+
+        /**
+         * Assign the role to each user/group
+         *
+         * @method _assignRoleToEachAssignee
+         * @protected
+         * @param {EventFacade} e
+         * @param {Role} role
+         * @param {Function} end
+         */
+        _assignRoleToEachAssignee: function (e, role, end) {
+            var userService = this.get('capi').getUserService(),
+                service = this,
+                roleAssignInputStruct;
+
+            Y.Array.each(e.selection, function (struct) {
+                // TODO assign with limitation
+                var limitation = null,
+                    contentTypeIdentifier = struct.contentType.get('identifier');
+
+                //var limitation = { // DUMMY SECTION DATA
+                //    "_identifier": "Section",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/sections/1",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }, {
+                //            "_href": "/api/ezp/v2/content/sections/4",
+                //            "_media-type": "application/vnd.ez.api.Section+json"
+                //        }]
+                //    }
+                //};
+                //var limitation = { // DUMMY SUBTREE DATA
+                //    "_identifier": "Subtree",
+                //    "values": {
+                //        "ref": [{
+                //            "_href": "/api/ezp/v2/content/locations/1/5/14",
+                //            "_media-type": "application/vnd.ez.api.Location+json"
+                //        }]
+                //    }
+                //};
+
+                // Convert hrefs to IDs, required by API
+                if (limitation) {
+                    Y.Array.each(limitation.values.ref, function (ref, i, refs) {
```

Comment:
since `limitation` is always null right now, this is dead code so you can remove it and this is where you should the todo comment with some explanation and a link to the corresponding JIRA issue.